src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```
@@ -792,27 +817,61 @@ private static List<CompletionResult> GetParameterCompletionResults(string param
                             $"-{alias}{colonSuffix}",
                             alias,
                             CompletionResultType.ParameterName,
-                            parameterType + alias));
+                            parameterType + alias + helpMessage));
                     }
                 }
             }
 
             return result;
         }
 
+#nullable enable
+        /// <summary>
+        /// Gets the help message text for the parameter attribute.
+        /// </summary>
+        /// <param name="attr">The attribute to check for the help message.</param>
+        /// <param name="assembly">The assembly to lookup resources messages, this should be the assembly the cmdlet is defined in.</param>
+        /// <returns>The help message of the parameter or <c>null</c> if none is defined.></returns>
+        private static string? GetParameterHelpMessage(
+            ParameterAttribute attr,
+            Assembly? assembly = null)
+        {
+            if (!string.IsNullOrWhiteSpace(attr.HelpMessage))
+            {
+                return attr.HelpMessage.Trim();
+            }
+
+            if (assembly is null || string.IsNullOrWhiteSpace(attr.HelpMessageBaseName) || string.IsNullOrWhiteSpace(attr.HelpMessageResourceId))
+            {
+                return null;
+            }
+
+            try
+            {
+                return ResourceManagerCache.GetResourceString(assembly, attr.HelpMessageBaseName, attr.HelpMessageResourceId)?.Trim();
+            }
+            catch (Exception e)
+            {
+                return $"Failed to find the help message: {e.Message}";
```

We never throw in the file.
```suggestion
```

---

As in `ResourceManagerCache.GetResourceString` shouldn't throw or are you trying to say something else? Because that method can throw if the base name is invalid.

```powershell
Function Test-Function {
    param (
        [Parameter(HelpMessageBaseName="invalid", HelpMessageResourceId="SomeId")]
        $Foo
    )
}

(TabExpansion2 'Test-Function -Foo').COmpletionMatches.ToolTip
# [Object] Foo - Failed to find the help message: Could not find the resource ...
```

I should probably write a test for this scenario.

---

Oops, I mean we should return null. We ignore all exceptions in the file.

---

Done.